html/js/guiutils/ContentEditable.js

```diff
@@ -0,0 +1,181 @@
+GTE.UI.Widgets = (function (parentModule) {
+    "use strict";
+
+    /**
+    * Creates a new ContentEditable object.
+    * @class
+    */
+    function ContentEditable(x, y, growingOfText, text) {
+        // Foreigns are needed in order to insert normal HTML elements within a
+        // SVG object. The idea behind the ContentEditable object is to introduce
+        // a div with the contenteditable attribute on true so that the user can
+        // edit the text inside the div as a normal input field.
+        this.myforeign = document.createElementNS('http://www.w3.org/2000/svg', 'foreignObject');
```

Comment:
Create an issue to check compatibility of foreignobjects with mobile browsers and IE. According to [this](https://developer.mozilla.org/es/docs/Web/SVG/Element/foreignObject#Browser_compatibility) there may be issues (it may not work at all in some browsers) and we may have to do something about it (possibly just warning somewhere that GTE is compatible only with chrome and firefox should be enough..)